File: Meseta_Verda.Domain/Entities/Producto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace Meseta_Verda.Domain.Entities
{
    public class Producto
    {
        [Key]
        public int IdProducto { get; set; }
        
       
        public int IdCategoria { get; set; }
        public int IdProveedor { get; set; }
        
   
        [Required]
        [MaxLength(200)]
        public string Nombre { get; set; } = null!;
        
        public string? Descripcion { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string UnidadMedida { get; set; } = null!;

       
        public Categoria Categoria { get; set; } = null!;
        public Proveedor Proveedor { get; set; } = null!;
        
      
        public ICollection<InventarioProveedor> Inventarios { get; set; } = new List<InventarioProveedor>();
        
    }
}

File: Frontend/agrotrade_frontend/lib/models/Productor/catalogo/producto.dart
class Producto {
  final int idProducto;
  final int idCategoria;
  final int idProveedor;
  final String nombre;
  final String? descripcion;
  final String unidadMedida;

  const Producto({
    required this.idProducto,
    required this.idCategoria,
    required this.idProveedor,
    required this.nombre,
    this.descripcion,
    required this.unidadMedida,
  });

  factory Producto.fromJson(Map<String, dynamic> json) {
    return Producto(
      idProducto: json['idProducto'] as int,
      idCategoria: json['idCategoria'] as int,
      idProveedor: json['idProveedor'] as int,
      nombre: json['nombre'] as String? ?? '',
      descripcion: json['descripcion'] as String?,
      unidadMedida: json['unidadMedida'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idProducto': idProducto,
      'idCategoria': idCategoria,
      'idProveedor': idProveedor,
      'nombre': nombre,
      'descripcion': descripcion,
      'unidadMedida': unidadMedida,
    };
  }
}
